Makefile

```diff
@@ -121,4 +121,4 @@ test-coverage: test-go coverage-report
 
 coverage-report:
 	@go install github.com/vladopajic/go-test-coverage/v2@v2.10.1
-	@go-test-coverage --config .testcoverage.yml
\ No newline at end of file
+	@go-test-coverage --config .testcoverage.yml && ([ $$? -eq 0 ] && exit 0) || exit 0
```

Comment:
I don't think this is a good idea, you are polluting the makefile with side effects to make github actions happy. This line returning a non 0 value is the expected behavior of any faulty test, any other CI script consuming this command would expect so.

Reply:
I totally agree with Luis. This is not a good solution.

Reply:
The current problem is that we need to do some exit code handling in `go.yml`.
I will elaborate. We need to:
- run `make coverage-report`, and we want the exit code of this command to dictate whether the step failed or not
- we want to save the output of the previous command to an env var or file, so that the next step can comment it
- we want to comment the output of previous step but only if it failed (this is doable with `if:failure()`)

The problem is in the first step, which should do the first 2 actions, run the comand, save the output, and report whether the command failed or not. But if the make command failed, then it will not execute the following command (which saves the output into an env var or file.

Maybe there is a more elegant manner of doing this, but (as far as I know) this can be done saving the exit code of the first command, and using it after the output text was saved. 

However, this requires more research into github workflows and time I think now should be spent in other tasks, so I will remove it all, just leave the basic, it fails if command fails, pass if success, and we will have to look into the build log to see what went wrong. 

Maybe in the future we can resume this research task and find a more elegant way of doing this.

Reply:
I am afraid that the Makefile is more important than the CI scripts: The expected behavior of a failing step is to return non 0. This is very important because any scripting is built upon this principle. 
The problem you are experiencing is that you need to make actions on failure. This needs to be a solution provided in the CI script and not in the makefile. 

I am afraid that if we can not solve this now, we need to give up on posting messages until we find the time to learn how to do it properly.